### PR TITLE
Clarify heap setting in Docker docs

### LINKED ISSUE #42660: Are Java memory options still needed for docker?
I was directed to this repo in https://github.com/elastic/elasticsearch-docker/issues/241

## Feature Description

Remove excess Java memory options to make compatible with modern Docker + Java workflow

https://github.com/elastic/elasticsearch-docker/blob/master/.tedi/template/docker-compose.yml.j2#L53

could instead be `-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap` if you use supporting JVM with container support.

Alternatively  `-XX:+UnlockDiagnosticVMOptions` promises to also adjust CPU to respect cgroup / isolation environments, and appears from https://github.com/docker-library/openjdk/issues/245 that OpenJDK official docker images support the backport to Java 8

The TLDR is that this tells docker to use the container, not host config to set sane limits.

Oracle docs here https://blogs.oracle.com/java-platform-group/java-se-support-for-docker-cpu-and-memory-limits

Blog post https://royvanrijn.com/blog/2018/05/java-and-docker-memory-limits/

Happy to PR & run some tests if you're up for one.

### PR DESCRIPTION
**Pull request by dliappis**
Add note in the Docker docs that even when container memory is limited,
we still require specifying -Xms/-Xmx using one of the supported
methods.

Closes #42660

### CODE REVIEW

File: docs/reference/setup/install/docker.asciidoc [hunk 1]
```diff
@@ -372,6 +372,10 @@ published ports with `--publish-all`, unless you are pinning one container per h
 
 . Use the `ES_JAVA_OPTS` environment variable to set heap size. For example, to
 use 16GB, use `-e ES_JAVA_OPTS="-Xms16g -Xmx16g"` with `docker run`.
++
+--
+NOTE: You still need to configure the heap size even if you are https://docs.docker.com/config/containers/resource_constraints/#limit-a-containers-access-to-memory[limiting memory access] to the container.
```
Comment (DaveCTurner) [inline docs/reference/setup/install/docker.asciidoc]: Suggest linking to the "setting the heap size" page, and formatting for width:

```suggestion
NOTE: You still need to <<heap-size,configure the heap size>> even if you are
https://docs.docker.com/config/containers/resource_constraints/#limit-a-containers-access-to-memory[limiting
memory access] to the container.
```

**Review by DaveCTurner — COMMENTED** [review]
I suggested a change.

File: docs/reference/setup/install/docker.asciidoc (hunk 1 above)
Comment (dliappis) [reply docs/reference/setup/install/docker.asciidoc]: Thanks! Addressed in https://github.com/elastic/elasticsearch/pull/42754/commits/274917c804916c69a15fb9fcffffd7fb77aa5631

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by dliappis** [comment]
@lcawl Do you have any thoughts on how far back the backport should go?

**Comment by lcawl** [comment]
It looks like this step exists all the way back to 5.0. Unless it's quite problematic, however, I typically don't backport past 6.8 (or 6.5 max) these days.

**Comment by dliappis** [comment]
@lcawl Thanks! This one doesn't fall in the problematic areas, just good to know, so I think I'll go until 6.8 then.

**Comment by dliappis** [comment]
Backported:

7.x -> 00f01aaece6516c522ded37244ad9dd09cc56311
7.2 -> e3ecbe01644491603b3701f3497c4c0c117e47db
7.1 -> cfedbfaa1116af877c5b9eecbb2edf5941e962d9
7.0 -> 2f644ba25614f73bca71a831eece665bb324413b
6.8 -> https://github.com/elastic/elasticsearch/pull/42885

**Pull request merged**